client/src/components/form/EditForm.jsx

```diff
@@ -0,0 +1,331 @@
+    saveInfoToggle();
+  };
+
+  const saveInfoToggle = () => {
+    setSaveInfo(true);
+  };
+
+  const handleSignatureData = (urlData) => {
+    setLog({
+      ...log,
+      signature: urlData,
+    });
+  };
+
+  const handleChange = (event) => {
+    let inputValue = event.target.value;
+    setLog({
+      ...log,
+      [event.target.name]: inputValue,
+    });
+  };
+
+  return (
+    <Center>
+      <CardContainer>
+        <form onSubmit={handleSubmit}>
+          <FormGroup>
+            <InputField
+              onTextInputChange={handleChange}
+              name='date'
+              type='date'
+              value={log.date}
+              placeholder=''
+              errorMessage='Please enter a valid date!'
+              required={true}
+              pattern='(0[1-9]|1[0-9]|2[0-9]|3[01]).(0[1-9]|1[012]).[0-9]{4}'
+            >
+              <img src={cardDate} alt='Date' />
+              <p>
+                Date of your dive<span>*</span>
+              </p>
+            </InputField>
+            <InputField
+              onTextInputChange={handleChange}
+              name='time'
+              type='time'
+              value={log.time}
+              placeholder=''
+              errorMessage='Please enter a valid time'
+              required={true}
+              pattern='([01]?[0-9]|2[0-3]):[0-5][0-9]'
+            >
+              <img src={cardTime} alt='Time' />
+              <p>
+                Time you started to dive<span>*</span>
+              </p>
+            </InputField>
+            <InputField
+              onTextInputChange={handleChange}
+              name='duration'
+              type='text'
+              value={log.duration}
+              placeholder='[45]'
+              errorMessage='Please enter the length of your dive in minutes [Numbers only]'
+              required={true}
+              pattern='[-+]?[0-9]*[.,]?[0-9]+'
+            >
+              <img src={cardDuration} alt='Duration' />
+              <p>
+                Length of your dive<span>*</span>
+              </p>
+            </InputField>
+          </FormGroup>
+          <FormGroup backgroundcolor='var(--background-card-place)'>
+            <InputField
+              onTextInputChange={handleChange}
+              name='country'
+              type='text'
+              value={log.country}
+              placeholder='[Spain]'
+              errorMessage='Please enter a Country'
+              required={true}
+            >
+              <img src={cardCountry} alt='Country' />
+              <p>
+                Country of your dive<span>*</span>
+              </p>
+            </InputField>
+            <InputField
+              onTextInputChange={handleChange}
+              name='spot'
+              type='text'
+              value={log.spot}
+              placeholder='[Costa Brava]'
+              errorMessage='Please enter the dive site'
+              required={false}
```

Comment:
'Required' is a Boolean attribute. It assumes the value of true once it is present. Therefore setting it to a 'false' still makes it act as though it was true.
So if the required attribute is present it's true - if it's absent, it's false. You have to remove this one. :)